File: src/employeemanager/model/Manager.java
package employeemanager.model;

import employeemanager.exceptions.InvalidDataException;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Manager extends FullTimeEmployee {

    private String managementPosition;
    private final List<Employee> team = new ArrayList<>();

    public Manager(int employeeId,
                   String firstName,
                   String lastName,
                   LocalDate startDate,
                   LocalDate dateOfBirth,
                   double annualSalary,
                   int vacationDays,
                   String benefits,
                   String managementPosition) throws InvalidDataException {
        super(employeeId, firstName, lastName, startDate, dateOfBirth,
              annualSalary, vacationDays, benefits);
        this.managementPosition = managementPosition;
    }

    public int getTeamSize() {
        return team.size();
    }

    public void addTeamMember(Employee employee) {
        if (!team.contains(employee)) {
            team.add(employee);
            employee.setManager(this);
        }
    }

    public void removeTeamMember(Employee employee) {
        team.remove(employee);
        if (employee.getManager() == this) {
            employee.setManager(null);
        }
    }

    public List<Employee> getTeam() {
        return Collections.unmodifiableList(team);
    }
    
    @Override
    public void reportToManager() {
        System.out.println("[Manager] " + getFullName() + " reports to director/VP.");
    }
}

File: Part 3/EmployeeManager/models/Manager.cs
using System;
using System.Collections.Generic;

namespace EmployeeManager.Models
{
    public class Manager : FullTimeEmployee
    {
        public string ManagementPosition { get; set; }
        private readonly List<Employee> _team = new List<Employee>();

        public Manager(int employeeId, string firstName, string lastName, DateTime startDate, DateTime dateOfBirth, double annualSalary, int vacationDays, string benefits, string managementPosition) : base(employeeId, firstName, lastName, startDate, dateOfBirth, annualSalary, vacationDays, benefits, managementPosition)
        {
            ManagementPosition = managementPosition;
        }

        public int TeamSize => _team.Count;

        public void AddTeamMember(EmployeeManager employee)
        {
            if (!_team.Contains(employee))
            {
                _team.Add(employee);
                employee.Manager = this;
            }
        }

        public void RemoveTeamMember (EmployeeManager employee)
        {
            _team.Remove(employee);
            if (employee.Manager == this)
                employee.Manager = null;
        }

        public IReadOnlyLost<Employee> Team => _team.AsReadOnly();

        public override void ReportToManager() => Console.WriteLine($"[Manager] {FullName} reports to director");
    }
}